### DISCUSSION TITLE
Ser Back/Front-end, ou Desenvolvedor Web?

### DISCUSSION THREAD

**Discussion by mathvbarone** [frontendbr/forum#655] [Discussão]
Pessoal, já rolaram varias [discussões ](https://github.com/woliveiras/front-end-career/blob/master/translations/pt-br/README.md)a respeito do futuro da profissão "Desenvolvedor Front-End". Alguns acham que a profissão vai se dividir em Front **End Engineer** e **Front End UI Developer**, outros acham que o futuro é não haver mais a separação "Back", e "Front". 

Enfim, meu questionamento é: Tentando projetar para daqui há alguns anos, qual seria o melhor caminho?
 Se especializar SÓ em front, ou SÓ em back, ou tentar se manter sempre no meio da balança?

Vou dar um exemplo real para contextualizar: Imagina um cara que tem uma noção de como funciona a web, mas esta querendo começar hoje. Será que o caminho é realmente ele escolher um lado, ou seria melhor ele tentar se aprofundar nos dois lados ao mesmo tempo? Ou, outro exemplo, um front quer estudar uma nova tecnologia, é uma boa ideia ele aprender uma linguagem de back, ou é melhor ele se aprofundar mais ainda no front?

Eu sei que essa pergunta já foi feita antes, mas eu queria saber a respeito do futuro da profissão, a tendência é dividir, ou somar?

**Comment by caioincau** [comment]
Eu acredito em um mundo cada vez mais especializado, logo acredito que faça sentido ser "só front", o Front moderno exige uma gama de conhecimentos como UI e acessibilidade, além das skills de código.

Agora a longo prazo, talvez alguma ferramenta consiga nos substituir, mas alguém ainda vai precisar manusear essa ferramenta, certo? Enfim, qualquer coisa além disso é especulação, o que sei é que hoje, está complicado contratar bons devs FE.

**Comment by mathvbarone** [comment]
@icaioincau Na minha opinião a profissão Dev FE já formou um universo tão grande, que eu pouco tempo ela mesmo vai se tornar um "Sub level", que vai se dividir e várias direções. Porém, pra quem esta começando hoje, conhecer o que rola por baixo dos panos de ponta a ponta é de um valor inestimável. É dai que surge a dúvida ahahahahaha

**Comment by clucasalcantara** [comment]
Então cara, a voracidade e velocidade em que a nossa área se atualiza em não consigo enxergar como uma pessoa que está começando possa escolher uma área, na minha concepção ela deveria ter um embasamento muito forte e buscar na caminhada dela o caminho que mais a agrada, mesmo que continue estudando coisas que não tenham nada a ver com a sua stack atual.

Até pq de nada adiantar vc saber webpack 3, Preact e SSR se você não souber encaixar isso no projeto ou gerar um diferencial pro que você está desenvolvendo no uso destas ferramentas.

**Comment by mathvbarone** [comment]
@clucasalcantara Eu concordo com você. Eu acho que ANTES de escolher pra onde seguir, a pessoa deve ter um embasamento do que ela esta fazendo. Se não acaba sendo um tiro no escuro.

**Comment by felipefialho** [comment]
Citando eu [mesmo nesse texto](https://medium.com/@lfeh/o-inicio-o-fim-e-o-meio-do-desenvolvimento-front-end-dfc5a123b90f), hehe

> Developers cada vez mais precisam entender e gostar de UX e participar ativamente da concepção de um produto. Sua percepção e olhar são essenciais para o sucesso deste.
Termos como: Web Developer, Web Designer, Web-qualquer-coisa, estão deixando de fazer sentido e estamos nos tornando Software Engineers, Software Developers, Product Engineers, Product Developers… ou qualquer outra palavra bonita que queira usar.

**Comment by mathvbarone** [comment]
@LFeh Mas se uma pessoa fosse começar hoje, vc aconselharia ele a seguir por um caminho especifico, ou a estudar o processo como um todo?

**Comment by felipefialho** [comment]
Ambas as coisas, hehe. 

O volume de informação é enorme, mas recomendaria se especializar de inicio em alguma coisa, e aos poucos entender todos os processo de desenvolvimento de softwares. 

Desde métodos ágeis, até desenvolvimento (Back-End, Front-End...), até UX.

Sempre vamos ter mais afinidade com uma coisa ou outra, mas em geral é importante ter uma boa noção de tudo.

**Comment by houstondapaz** [comment]
Concordo com o @clucasalcantara, a pessoa tem que criar uma base sólida de conhecimento, que é a lógica, saber como as coisas realmente funcionam.
Muitos simplesmente sabem fazer e não sabem como é feito, assim existem diversos "especialistas" que se a coisa não funcionar do jeito que ele aprendeu ele não sabe o que fazer.
Eu sempre busquei focar minha carreira em web (na época que eramos programadores e não existia separação entre back/front, no máximo tinha o designer e o programador) mas procurei saber como funcionava tudo, desde a renderização do html no browser, a conexão http, o estado no servidor e a conexão com o banco. Com isso consigo atuar em qualquer treta que acontece, mas... Foi uma looonga jornada e alguem que ta começando hoje precisa de um norte e eu falaria pra ele estudar a base (lógica de programação) e ir experimentando as areas para descobrir o que gosta de fazer, se especializar nela, mas não se bitolar SÓ nela.

TL;DR
Aprenda lógica de programação, se especialize em algo que goste, mas busque conhecer o processo todo.

**Comment by rubenmarcus** [comment]
Muita vaga pede conhecimento em Node.js, se você ver aqui no fórum de vagas, a grande maioria pede, que é considerado Back. Hoje em dia um Front sem conhecimento mínimo de Node.js e PHP (maioria pedem , algumas poucas pedem ASP.NET e Java ou Ruby),está fora do mercado.

A não ser que você mire o mercado de sites institucionais em temas de Wordpress, mas ai ao meu ver você não é um programador, você é um bom vendedor / marketeiro que sabe implementar um site. É um pouco diferente.

**Comment by woliveiras** [comment]
Sobre Node em vagas de Front End eu tenho uma opinião a compartilhar:

A maioria das vagas de Front que pedem Node (maioria, não todas) são somente para consumir APIs com o Node e devolver os dados para uma SPA ou mesmo para que a pessoa saiba fazer um SSR (assunto de outra issue que abri aqui no forum)...

Não é nenhum conhecimento avançado em Back End.

Se a pessoa sabe consumir recursos na Web com Ajax, vai pelo menos entender como funciona uma API com um ExpressJS da vida e se estudar alguns minutos consegue desenvolver uma aplicação bem simples para devolver dados para a aplicação SPA. Depois vai avançando nos estudos até que consiga criar algo mais decente.

O serviço pesado (tratar os dados e cuidar do tráfego e recursos de servidor), **normalmente**, está em grandes aplicações back end fazendo o processamento de grandes massas de dados e devolvendo filtrado para os WebServices em Node.

As aplicações web que uma pessoa de front cria com tecnologias web funciona mais como um proxy do que de fato como aplicação back end. - pega daqui, manda pra lá, pega daqui, faz um de-para, manda pra lá...

No Brasil nem temos tantas vagas (não significa que não temos) que são para fazer coisas tão avançadas, não precisa ter medo de se envolver no Back End achando que vai precisar criar o próximo algoritmo que vai ativar os recursos que lançam foguetes para o espaço.

Percebo que muitas vezes achamos que ter Node no requisito da vaga é algo que vai nos fazer sofrer no trabalho, mas não é não gente.

Acredito que o mesmo se encaixa para vagas de front end que pedem PHP, Ruby, Python... Não é nada tão avançado pra que quem tem conhecimento em programação client side não saiba mexer com. 

**Programação é programação, seja no Front End ou no Back End.**

A questão de dizer que não é tão avançado não é para minimizar os esforços da pessoa que vai trabalhar com isso, mas pelo fato de que não precisa sair da nossa área e virar full stack pra sobreviver a vagas de front que pedem conhecimento mínimo de back end.

Acho que nós, como **desenvolvedores de software**, não somente como front enders, deveríamos sim conhecer um pouco mais sobre como os produtos que trabalhamos estão no ar e entender como funciona um serviço que está na AWS, Cloud Platform, Azure, pelo menos saber o por que usar MySQL ou MariaDB e por que usar Mongo, DynamoDB, etc. Podemos conhecer de desenvolvimento web no geral e trabalhar com Front End...

Acho que não está escrito em pedra o que devemos ou não saber, podemos parar de limitar nossa vontade de aprender, pois estudar um pouco de back end não vai tirar todo o nosso tempo de estudo em front end.

Acredito que pensar em ser somente front end ou somente front end acaba limitando nossa força de vontade em aprender/conhecer um pouco de outras coisas. - lembrando que aprender não significa dominar 100% da outra coisa que você se interessa. Eu posso me interessar por psicologia, mas não por isso eu faço uma faculdade da área.

**Comment by woliveiras** [comment]
Acho que falei tanto que fugi do foco principal do tópico: eu acredito que o futuro é somar e voltar tudo a ser para desenvolvedores de software ou engenheiros de software e pronto.

A área de desenvolvimento é um verdadeiro vai e vem... Antigamente era tudo renderizado nos servidores e depois foi separado, hoje já temos a renderização no servidor de novo... Antigamente se proibiu colocar HTML, CSS e JS no mesmo local, hoje temos single files templates e outras coisas do tipo... É um vai e vem desgraçado.

Vai voltar a ser tudo junto e depois vai separar de novo, pra depois voltar a juntar. É nisso que acredito.

Ou se aprende as bases, como o @clucasalcantara e o @houstondapaz falaram, ou sofre um pouco no mercado de trabalho por conta dessas mudanças.

**Comment by mathvbarone** [comment]
@woliveiras Cara, seu comentário foi excelente, muito obrigado!

Falando um pouco do que eu pretendo fazer....primeiro estudar a fundo as 3 linguagens front (html, css e js), e depois estudar um pouco o que se passa no server-side. Particularmente não é uma área que me agrade, mas eu acho extremamente válido ter esse conhecimento, como você mesmo disse, nem que seja para utilizar apenas o "básico" disso.

O foda é arranjar tempo pra estudar tudo....queria conseguir absorver as coisas por osmose ahahahahahah.

Como vocês se organizam nesse quesito de foco no estudo?

**Comment by houstondapaz** [comment]
Andar de transporte público rs
Pelo menos pra mim eu ganho 2 horas por dia para ler artigos/livros e ouvir podcasts
Ai vc tem que organizar sua trilha de estudos, saber quais as tecnologias quer aprender e ir do mais simples para os assuntos mais complexos...

ahhh um kindle ajuda bastante

**Comment by mathvbarone** [comment]
@houstondapaz Cara, eu faço o mesmo ahahahahaha eu ja escutei todos os DEV na Estrada, PodProgramar, Hipsters.Tech...tem mais alguns p indicar?

**Comment by woliveiras** [comment]
Tem alguns podcasts em inglês que eu escuto pra treinar um pouco do idioma e aprender com uns caras de fora, @mathvbarone:

* https://softwareengineeringdaily.com/category/podcast/
* http://frontendhappyhour.com/
* http://bikeshed.fm/
* http://www.codingblocks.net/
* http://thedebuglog.com/
* https://changelog.com/podcast
* https://devchat.tv/js-jabber
* https://talkpython.fm/
* https://learntocodewith.me/podcast/
* http://www.codenewbie.org/podcast
* https://www.arresteddevops.com/
* http://www.fullstackradio.com/
* http://www.programmingthrowdown.com/

A maioria das vezes eu não entendo quase nada, mas pelo pouco que entendo da pra aprender muito nesses casts ai. Se você estiver bem do inglês, vai aprender demaaais!

BRs ainda tem esses aqui:

* https://www.lambda3.com.br/tag/podcast/
* https://www.concrete.com.br/category/prodcast/?categoryId=21

**Comment by woliveiras** [comment]
Sobre os estudos, a única coisa que digo é: **tente não querer aprender de tudo e ser ninja em tudo de uma só vez**.

A pressa faz com que a gente não aprenda e ainda por cima tira totalmente nosso foco.
